### PR TITLE
[Docs revamp] consolidated all export related tasks in one section of howtos

### PR DESCRIPTION
**Pull request by Rajakavitha1**
Description: Please include the following in your PR description:
* Consolidated all export related information and rewrote the content in the how to framework.

Note: Create and edit docs in the `main` branch when you can, so that we only cherry-pick out of `main`, not into `main`.
-->
_ i have remved references to check queue in the export content because as Rahul confirmed "Yes, that's correct. We cannot query queued/running tasks on any v21.03.1 (and beyond) EE or OSS builds. (edited) "

Fixes: https://github.com/dgraph-io/dgraph-docs/issues/352 https://github.com/dgraph-io/dgraph-docs/issues/314

### CODE REVIEW

File: content/howto/exportdata/export-data.md [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+title = "Export data from Dgraph"
+keywords = "export, data, self hosted"
+[menu.main]
+    parent = "exportdata"
+    weight = 3
++++
+
+As an `Administrator` you can export data on all nodes, configure the Alpha server, specify the export format, export to an object store, disable HTTPS for exports, and encrypt exports
+
+## Export data on all nodes
```
Comment (rderbier) [inline content/howto/exportdata/export-data.md:11]:
```suggestion
## Export data using  the GraphQL admin  endpoint
```

File: content/howto/exportdata/export-data.md [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+
+You can export all nodes by executing a GraphQL mutation on the `/admin` endpoint of an Alpha node ( `localhost:8080/admin`) using any GraphQL client.
```
Comment (rderbier) [inline content/howto/exportdata/export-data.md:13]:
```suggestion
You can export the entire data by executing a GraphQL mutation on the `/admin` endpoint of any Alpha node.
```

File: content/howto/exportdata/export-data.md [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+
+**Before you begin**:
+
+*  Specify a list or range of IP addresses allowed to initiate admin operations such as export using the command `dgraph alpha --security whitelist <IP address>`.
```
Comment (rderbier) [inline content/howto/exportdata/export-data.md:17]: I would remove this line. It's about securing DB access and is explained elsewhere.

File: content/howto/exportdata/export-data.md [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+*  Ensure that there is sufficient space on disk to store the
+export.
+
+This mutation triggers an export for all Alpha groups of the cluster.
```
Comment (rderbier) [inline content/howto/exportdata/export-data.md:21]:
```suggestion
This mutation triggers the export.
```

File: content/howto/exportdata/export-data.md [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+mutation {
+  export(input: {}) {
+    response {
+      message
+      code
+    }
+  }
+}
+```
+The export data of the group:
```
Comment (rderbier) [inline content/howto/exportdata/export-data.md:33]: we should better explain what's going on.
Maybe 
This mutation triggers the export : the Alpha node leader of each Alpha group will write an export file for the group in the directory that was specified by the --export parameter used when launching the node.
Depending on you Dgraph configuration you may obtain several files (one per Alpha group).

We recommend that you copy the files from the alpha server nodes to a safe place when the export is finished.

Comment (rderbier) [reply content/howto/exportdata/export-data.md:33]: BTW how do we know that the export is done ?

File: content/howto/exportdata/export-data.md [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+
+* in the Aplha instance is stored in the Alpha.
+* in every other group is stored in the Alpha leader of that group.
+
+You need to retrieve the right export files from the Alpha instances in the
+cluster. Dgraph does not copy all files to the Alpha that initiated the export.
+
+
+## Configure Dgraph Alpha server nodes
```
Comment (rderbier) [inline content/howto/exportdata/export-data.md:42]: I would just reference the config page and put it as a pre-req step 
" Before exporting : check that you know the export directories of the alpha server nodes, see <config ref>".

File: content/howto/exportdata/export-data.md [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+
+Each Dgraph Alpha leader for a group writes output as a gzipped file to the export
+directory specified through the `--export` flag (defaults to an **export** directory). If any of the groups fail, the entire export process is considered failed and an error is returned.
```
Comment (rderbier) [inline content/howto/exportdata/export-data.md:45]: I would add a section "Attention point".
If one of the Alpha leader does not have enough disk space or encounters an issue, the entire export is considered failed and an error is returned"

File: content/howto/exportdata/export-data.md [hunk 8]
```diff
@@ -0,0 +1,296 @@
...
+    response {
+      message
+      code
+    }
+  }
+}
+```
+
+## Export to an object store
+You can export to an S3 or MinIO object store by specifying a URL in the `destination` input field.
```
Comment (rderbier) [inline content/howto/exportdata/export-data.md:97]: We should remove initial reference to MinIO. MinIO is just a way to store to Google or Azure and we will soon remove the need to use MinIO. The feature is to save to Azure Blob Storage and Google Cloud Storage
```suggestion
You can export to an AWS S3, Azure Blob Storage or Google Cloud Storage.
```

**Review by rderbier — CHANGES_REQUESTED** [review]
Lot's of work here !

Need some reviews on the export for Dgraph (on prem) as this part is not clear enough for me.

File: content/deploy/dgraph-administration.md [hunk 9]
```diff
@@ -137,330 +137,15 @@ $ curl -H 'X-Dgraph-AuthToken: <authtokenstring>' localhost:8080/alter -d '{ "dr
 To fully secure alter operations in the cluster, the authentication token must be set for every Alpha node.
 {{% /notice %}}
 
+## Export database
```
[the remark this answers is not in the document]
Comment (Rajakavitha1) [reply content/deploy/dgraph-administration.md:140]: As a user I would export it from Dgraph to import it else where. However, it is important to list import section in the where to next section. The reason I did not provide the link to import is dont want the user to hop between two guides when performing the action of export.

File: content/enterprise-features/encryption-at-rest.md [hunk 10]
```diff
@@ -112,11 +112,11 @@ If the Alpha server restarts, the `--encryption key-file` or the `--vault` super
 
 ## Turn off Encryption
 
-You can use [live loader]({{< relref "live-loader.md" >}}) or [bulk loader]({{< relref "bulk-loader.md" >}}) to decrypt the data while importing.
+You can use [live loader]({{< relref "/deploy/fast-data-loading/live-loader.md" >}}) or [bulk loader]({{< relref "/deploy/fast-data-loading/bulk-loader.md" >}}) to decrypt the data while importing.
```
[the remark this answers is not in the document]
Comment (Rajakavitha1) [reply content/enterprise-features/encryption-at-rest.md:115]: Yes!!!! Does not make sense to add a task about things to consider when importing in the export section. I will remove it.

File: content/howto/exportdata/export-data.md (hunk 6 above)
Comment (Rajakavitha1) [reply content/howto/exportdata/export-data.md:42]: Yes!!! I agree.

File: content/howto/exportdata/export-data.md (hunk 5 above)
Comment (Rajakavitha1) [reply content/howto/exportdata/export-data.md:33]: Good question @MichelDiz  and @rarvikar  Please provide the steps to confirm if the export was successful.

File: content/enterprise-features/encryption-at-rest.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (Rajakavitha1) [reply content/enterprise-features/encryption-at-rest.md:115]: @rderbier  the only change i made was for the rel ref. We can take this up during the enterprise-feature rewrite.

File: content/howto/exportdata/export-data.md (hunk 5 above)
Comment (MichelDiz) [reply content/howto/exportdata/export-data.md:33]: The terminal output will show "export finished" something like this. And there is a GraphQL Query to check the status.

File: content/howto/exportdata/export-data.md (hunk 7 above)
Comment (rderbier) [reply content/howto/exportdata/export-data.md:45]: Action is to verify the disk space I think

Comment (Rajakavitha1) [reply content/howto/exportdata/export-data.md]: done!!

File: content/howto/exportdata/export-data.md (hunk 6 above)
Comment (Rajakavitha1) [reply content/howto/exportdata/export-data.md]: done!!!

File: content/howto/exportdata/export-data.md (hunk 5 above)
Comment (Rajakavitha1) [reply content/howto/exportdata/export-data.md:34]: added the response

File: content/howto/exportdata/export-data.md (hunk 3 above)
Comment (Rajakavitha1) [reply content/howto/exportdata/export-data.md]: removed.

File: content/howto/exportdata/export-data-cloud.md [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+
+   Depending on the format that you chose to create an export, three files are generated.
+   
+{{% notice "note" %}}
+Ensure that you download these files as soon as possible because the signed URLs to download these files expire after 48 hours from the time they were generated. You can use `curl -O <SIGNED_URL>` to download the files to the current directory.
+{{% /notice %}}   
+
+### Exporting data from Dgraph Cloud programatically
+
+You can also export data from Dgraph Cloud programatically using the Dgraph Cloud API. For more information, see [Cloud API documentation](https://dgraph.io/cloud-docs/blob/master/content/cloud-api/backup.md#export-data).
```
Comment (rderbier) [inline content/howto/exportdata/export-data-cloud.md:85]:
```suggestion
You can also export data from Dgraph Cloud programatically using the Dgraph Cloud API. For more information, see [Cloud API documentation](https://dgraph.io/docs/cloud/cloud-api/backup/#export-data).
```

**Review by rderbier — APPROVED** [review]

**Review by MichelDiz — APPROVED** [review]

**Pull request merged**